backend/src/resources/todos/todo.router.ts

```diff
@@ -1,23 +1,46 @@
-import { NextFunction, Request, Response, Router } from 'express';
+import { Request, Response, Router } from 'express';
 import { Todo } from './todo.model';
 
-const todos: Todo[] = [];
+let todos: Todo[] = [];
 const todosRouter = Router();
 
-todosRouter.get('/', (req: Request, res: Response, next: NextFunction) => {
-  res.json({
-    message: 'Todo: Implement',
-  });
+todosRouter.get('/', (req: Request, res: Response) => {
+  res.json(todos);
 });
 
-todosRouter.post('/', (req: Request, res: Response, next: NextFunction) => {
-  const todo: Todo = {
-    id: Math.random().toString(36).substring(2, 7),
-    description: req.body.description,
-    isDone: false,
-  };
-  todos.push(todo);
-  res.json(todo);
+todosRouter.delete('/:id', (req: Request, res: Response) => {
+  if (todos.findIndex(todo => todo.id === req.params.id) !== -1) {
+    todos = todos.filter(todo => todo.id !== req.params.id);
+    res.json({ status: true });
+  } else {
+    res.json({ status: false });
+  }
+});
+
+todosRouter.patch('/:id', (req: Request, res: Response) => {
+  if (!req.body.isDone) {
+    throw new Error('Missing isDone in body');
+  } else {
+    const index: number = todos.findIndex(todo => todo.id === req.params.id);
+    if (index !== -1) {
+      todos[index].isDone = req.body.isDone;
+      res.json(todos[index]);
+    }
+  }
+});
+
+todosRouter.post('/', (req: Request, res: Response) => {
+  if (!req.body.description) {
```

Comment:
Like the check for the patch endpoint, this one also has a similar issue. You should check that:

1. It is ensured that `description` is sent in the body
2. `description` is a `string` with length greater than `0`

If either of the conditions is not met, the client must get an appropriate error message.